### PR TITLE
memory-id: drop condition which never hit

### PR DESCRIPTION
**Pull request by yuwata**
cc @hadess.

### CODE REVIEW

**Comment by bluca** [comment]
Could you please add a line or two to the commit message, explaining why it's never hit? I'm not familiar with DMI so looking at the diff it's not immediately obvious to me. Thanks!

File: src/udev/dmi_memory_id/dmi_memory_id.c [hunk 1]
```diff
@@ -601,11 +599,7 @@ static int smbios3_decode(const uint8_t *buf, const char *devmem, bool no_file_o
         if (!verify_checksum(buf, buf[0x06]))
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Faied to verify checksum.");
 
-        offset = QWORD(buf + 0x10);
-        if (!no_file_offset && (offset >> 32) != 0 && sizeof(int64_t) < 8)
```
Comment (bluca) [inline src/udev/dmi_memory_id/dmi_memory_id.c:605]: is it because int64_t is always guaranteed to be 8 bytes large?

**Comment by bluca** [comment]
> Could you please add a line or two to the commit message, explaining why it's never hit? I'm not familiar with DMI so looking at the diff it's not immediately obvious to me. Thanks!

Thank you! In my defense, it's just morning and I hadn't had my tea yet :-)

**Comment by yuwata** [comment]
I also created #18016. Let's wait for @hadess's comment.

**Comment by hadess** [comment]
> I also created #18016. Let's wait for @hadess's comment.

I prefer https://github.com/systemd/systemd/pull/18016
Do you want to close this?

**Pull request closed without merging**